plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -249,7 +250,7 @@
     public static final String CHANGE_COLUMN_OPERATION = "CHANGE COLUMN";
     public static final String ISOLATION_LEVEL = "WriteSerializable";
     private static final int READER_VERSION = 1;
-    private static final int WRITER_VERSION = 2;
+    private static final int WRITER_VERSION = 3;
```

Comment:
I think after this change we need to split this into two different properties. There's the required writer version that we support writing to, and then there's the required writer version that tables we create need.

We can write to table version 3 (as long as they don't have `check` constraints)
We create tables that need version 2

Once we can set the `delta.checkpoint.writeStatsAs*` property, we'd bump the needed version to 3 when they are set. Does that make sense?

Reply:
Agreed, I was thinking similar thing. Reverted the bump since this can be follow-up PR.